proteuslib/unit_models/reverse_osmosis_0D.py

```diff
@@ -910,17 +919,20 @@ def initialize(blk, initialize_guess=None, state_args=None, outlvl=idaeslog.NOTS
                          package(s) to provide an initial state for the inlet
                          feed side state block (see documentation of the specific
                          property package) (default = None).
-            outlvl : sets output level of initialization routine
+            outlvl_init : sets output level of InitLogger, init_log (default= idaeslog.INFO)
```

Comment:
This may be a problem in some cases, depending on how the core IDAES code was written. Some unit models (or control volumes) may assume that `outlvl` is defined in all `initialize` methods, which will cause failures here.

Why did want to separate these two logger levels?

Reply:
I set a second one specifically for the solver, in case someone wanted to set only that to DEBUG while setting whatever they want for the `init_log`. I suppose I was just riffing, thinking it would add further flexibility for troubleshooting initialization issues, but if you think it'll cause problems, I can just go back to using the one logger level.

Reply:
Actually - it generally won't be a problem as this is a unit model itself (I am too used to thinking from the bottom up). However, it could be an issue for someone using the automatic sequential decomposition tool on a flowsheet who wants to set the `outlvl` option. Most core models only have `outlvl`, so these models will be an exception which would mean you need to catch and real with this exception when calling `initialize`.

The reason we only had one `outlvl` was that we figured that if you wanted to see the full solver output, you probably also needed most of the debug information to put it in context. With the two separated, you could in theory set the solver output to debug and the general output to none and only see the solver output, but without any context to tell you what each solve was doing.

Reply:
I think I recall the reason why I did this now; I noticed that several initialization routines don't use `outlvl` for the solver_log. Instead, the logger level typically seems to be hard-coded to DEBUG as follows:
`with idaeslog.solver_log(solve_log, idaeslog.DEBUG) as slc:` 

So I figured--why not give the user the option to set this logger level rather than having it be hard-coded (even though setting to DEBEUG is perfectly fine and logical)? 

Additionally, could the potential issue with flowsheets be resolved by 

1. renaming `outlvl_int` to `outlvl` and then setting `outlvl_solve=None` and 
2. include a conditional within the `initialize()` function that sets `outlvl_solve` to some default logger level if it is `None`? 

Overall, it seems less clean to do this, and I'm planning to change things back to how they were, as you're suggesting. I just wanted to think more about the hypothetical remedy before doing so.

Reply:
You could do that, but I would be inclined to just have the single `outlvl`. I don't see too much need for having separate output levels for the solver and the general logger. There generally isn't too much debug information anyways, and if you are looking at solver outputs during initialization you are probably trying to work out why it didn't work, which means that any debug messages are probably useful to you too.

I noted the issue with hard coding the solver output level whilst reviewing this, and my response would have been that they should have used `outlvl` rathe than hard coding it.